libs/sql-schema-describer/src/mysql.rs

```diff
@@ -578,17 +587,33 @@ fn get_column_type_and_enum(
     full_data_type: &str,
     precision: Precision,
     arity: ColumnArity,
+    default: Option<&Value>,
 ) -> (ColumnType, Option<Enum>) {
+    // println!("Name: {}", column_name);
     // println!("DT: {}", data_type);
     // println!("FDT: {}", full_data_type);
     // println!("Precision: {:?}", precision);
+    // println!("Default: {:?}", default);
+
+    let is_tinyint1 = || extract_precision(full_data_type) == Some(1);
+    let invalid_bool_default = || match default {
+        None => false,
+        Some(param_value) => match param_value.to_string() {
+            None => false,
+            Some(x) if x == "NULL" => false,
+            Some(default_string) => match parse_int(&default_string) {
+                Some(PrismaValue::Int(1)) => false,
+                Some(PrismaValue::Int(0)) => false,
+                None => false,
+                _ => true,
+            },
+        },
+    };
```

Comment:
I think this closure could be simplified to something like

```
    default
        .as_ref()
        .and_then(|default| default.to_string())
        .filter(|default_string| default_string != "NULL")
        .and_then(|default_string| parse_int(&default_string))
        .filter(|default_int| default_int != 0 && default_int != 1)
        .is_some()
```

It's a matter of preference though, no need to change this.

Reply:
Yeah, I like the functional style of reducing it to invalids! This is what worked: 

default
            .and_then(|default| default.to_string())
            .filter(|default_string| default_string != "NULL")
            .and_then(|default_string| parse_int(&default_string))
            .filter(|default_int| *default_int != PrismaValue::Int(0) && *default_int != PrismaValue::Int(1))
            .is_some()